src/tools/align/utils/useSequenceInfo.ts

```diff
@@ -22,63 +26,127 @@ export type SequenceInfo = {
+    const extracted = extractAccession(processedSequence.name);
+    if (extracted?.accession) {
+      processedSequences.push({
+        ...processedSequence,
+        ...extracted,
+      });
+    }
+  }
+  return processedSequences;
+};
+
+const getEndpoint = (
+  processedArray: ParsedSequenceAndFeatures[],
+  namespace: Namespace
+) =>
+  getAccessionsURL(
+    processedArray
+      .filter((p) => p.namespace === namespace)
+      .map((p) => p.accession),
+    { namespace, facets: null }
+  );
 
-// TODO: handle non-UniProtKB entries too!
-// Need to separate the entries, call the 3 accessions endpoints, and merge data
-// https://www.ebi.ac.uk/panda/jira/browse/TRM-26756
 const useSequenceInfo = (rawSequences?: string): SequenceInfo => {
   const processedArray: ParsedSequenceAndFeatures[] = useMemo(
-    () =>
-      sequenceProcessor(rawSequences || '')
-        .map((processed) => ({
-          ...processed,
-          accession: extractAccession(processed.name),
-        }))
-        // ensures we managed to extract an accession from header, otherwise discard
-        .filter(hasAccession),
+    () => processSequences(rawSequences),
     [rawSequences]
   );
 
-  const endpoint = getAccessionsURL(
-    processedArray.map((processed) => processed.accession),
-    { facets: null }
-  );
-  const { data, loading, error } =
-    useDataApi<SearchResults<UniProtkbAPIModel>>(endpoint);
+  const uniprotkbEndpoint = getEndpoint(processedArray, Namespace.uniprotkb);
+  const unirefEndpoint = getEndpoint(processedArray, Namespace.uniref);
+  const uniparcEndpoint = getEndpoint(processedArray, Namespace.uniparc);
 
-  const outputData = useMemo(() => {
-    const dataPerAccession = new Map(
-      data?.results.map((object) => [object.primaryAccession, object]) ?? []
-    );
+  const uniprotkbResults =
+    useDataApi<SearchResults<UniProtkbAPIModel>>(uniprotkbEndpoint);
+  const unirefResults =
+    useDataApi<SearchResults<UniRefAPIModel>>(unirefEndpoint);
+  const uniparcResults =
+    useDataApi<SearchResults<UniParcAPIModel>>(uniparcEndpoint);
 
+  const outputData = useMemo(() => {
+    const idToSequenceAndFeatures = new Map<
+      string,
+      {
+        sequence: string;
+        features?:
+          | UniProtkbAPIModel['features']
+          | UniParcAPIModel['sequenceFeatures'];
+      }
+    >();
+    for (const result of uniprotkbResults?.data?.results || []) {
+      idToSequenceAndFeatures.set(result.primaryAccession, {
+        sequence: result.sequence.value,
+        features: result.features,
+      });
+    }
+    for (const result of unirefResults?.data?.results || []) {
+      idToSequenceAndFeatures.set(result.id, {
+        sequence: result.representativeMember.sequence.value,
+      });
+    }
+    for (const result of uniparcResults?.data?.results || []) {
+      idToSequenceAndFeatures.set(result.uniParcId, {
+        sequence: result.sequence.value,
+        // features: result.sequenceFeatures,
```

Comment:
Is it expected to adapt the features as part of this task?

Reply:
That was not in my mind when creating the jira originally, but it would make a lot of sense to do that I believe, even though it's a bit more work.

1. UniProtKB and UniProtKB isoforms should be the same, so not a big problem here.
2. UniParc should be simple too as we have InterPro features in the data, although the shape might be different as the features in UniProtKB.
3. UniRef might be a bit more complex. The sequence is that of the representative, and the representative is either a UniProtKB (canonical or isoform), or a UniParc. Maybe the logic could be that of fetching the features from the existing logic in points 1 and 2, extracting the representative from the UniRef ID (UniRef<identity>_<representative>.